trieste/models/gpflow/models.py

```diff
@@ -973,6 +972,7 @@ def update(self, dataset: Dataset) -> None:
                 )
             ):  # only bother updating if points actually change
                 self._update_inducing_variables(new_inducing_points)
+                self.update_posterior_cache()
```

Comment:
@uri-granta this is not a minor change, previously update didn't change the posterior cache correctly. Didn't matter as long as we called optimize just after update. But unsure if there are other consequences...

Reply:
Shouldn't this be run even if `self._inducing_point_selector` is None (like in `SparseGaussianProcessRegression`)?

Regarding testing, is the issue that `test_gpflow_models_cached_predictions_correct` and the other posterior cache tests don't use a `SparseVariational` with an `inducing_point_selector`? If so, it might be useful to add a simple test that checks both this and `SparseGaussianProcessRegression`.

Reply:
I am not sure that it is necessary if you don't change your inducing points, in that case the only change is the data size which has no impact on the posterior.

We could play it safe and run it anyway, but I'm a bit unsure how much overhead this adds.